# RNA-seq - Bern-Cichlids team <Â°)))o><

## General Description & Disclaimer
This pipeline was designed to clean raw read files and map them back on a reference transcriptome. Be careful when using the scripts, everything is well formatted for the specific requirements of the KATAK cluster at Laval University. Follow the instructions step by step to produce the desired output, but if this pipeline is used on another cluster, there is absolutely no guarantee that it will work properly. You can always go back into the script files and change the code according to your needs.
<br>
## Step 1 - Copying and expanding raw files
**Description**<br>
First, you need to copy the files that are on the external hard-drive to your account on KATAK, in the "raw data" folder of the pipeline. Here's how you can perform this task:<br>

*Copying the files from external hard-drive to raw data folder using the "[rsync](https://en.wikipedia.org/wiki/Rsync)" utility. Open a terminal window and type the following code:*<br><br>
`rsync -avzP /Volumes/$HD/*.fastq.tar.gz ckavoekl@ibis.ulaval.ca:/home/ckavoek1/align_on_reference/02_raw_data`<br><br>
*NOTE*: $HD = name of your external hard-drive<br>

**1.1 First, go to the pipeline main folder**:<br><br>
`cd /home/ckavoekl/align_on_reference`<br><br>
**1.2 Submit the expand job**:<br><br>
`qsub 01_scripts/jobs/00_expand.job.sh`
<br>
## Step 2 - Trimming reads (PHRED score > 2)
**Description**<br>
This step will clean the reads according to a sequencing quality threshold, i.e. PHRED score greater than 2. Trimmomatic will discard the bases from both ends that show a sequencing quality below the user-defined threshold. In this case, the threshold is quite low (PHRED > 2). Agressive trimming leads to loss of important information, so the threshold is set low. Still from the main folder of the pipeline (i.e. /home/ckavoek1/align_on_reference), here's how you can perform this task:<br>

*2.1 Submit the trimming job:*<br><br>
`qsub 01_scripts/jobs/01_trimming_job.sh`<br><br>

**IMPORTANT NOTE - COMPLETING THE PIPELINE WITH A SUBSET OF THE DATA**<br>
* That last command line will take **ALL** of the FASTQ files found in '02_raw_data/', trim/clean them and place the output files (i.e. trimmed read files) into the folder for the next step, i.e. 03_trimmed.
* If you want to perform the analysis with only a subset of the data to practice or test the pipeline, here is what you could do:<br><br>

*From the main folder of the pipeline, create a folder in which you place all the read files except the few ones you want to work with*:<br><br>
`mkdir 02_raw_data/temp_samples # Creates the folder`<br>
`mv 02_raw_data/*.fastq 02_raw_data/temp_samples # Places all the read files in the temporary folder`<br>
`mv 02_raw_data/temp_samples/*92S518*.fastq 02_raw_data/ # Moves sample number 92S518 back in the raw_data directory`<br><br>
*And then submit the job*:<br><br>
`qsub 01_scripts/jobs/01_trimming.job.sh`

This will allow you to continue the pipeline with sample number 92S51 only.

## Step 3 - Aligning the reads on the reference
**Description**<br>
This step will align the reads on the reference transcriptome (coding sequences, CDS), in this case the Nile tilapia ([*Oreochromis niloticus*](http://www.ensembl.org/Oreochromis_niloticus/Info/Index)). Still from the main folder of the pipeline (i.e. /home/ckavoek1/align_on_reference), here's how you can perform this task:<br>

`qsub 01_scripts/jobs/02_bwa_mapping.job.sh`<br>

## Step 4 - Getting the count matrix
**Description**<br>
This step will allow you to produce a count matrix based on the SAM files generated by BWA during the mapping step. Still from the main folder of the pipeline (i.e. /home/ckavoek1/align_on_reference), here's how you can perform this task:<br>

`qsub 01_scripts/jobs/03_read_counts.job.sh`<br>
<br>
This last script should produce an output file called `samples.read-count.tsv` in the folder `05_read_counts`. This output file is the actual matrix that you need as the input file in edgeR or limma. ROWS = genes ("reference sequences"), COLUMNS = samples.<br>

## Step 5 - Annotation
**Description**<br>
There is no extensive annotation step in this pipeline since the tilapia genome is used as the reference. The cool thing is that an annotation file already exists. You can find it in:<br>

`02_raw_data/tilapia_genome/ensembl.annotation.txt`
<!---->
